pkg/podexecutor/staticpod.go

```diff
@@ -101,14 +102,17 @@ func (s *StaticPodConfig) APIServer(ctx context.Context, etcdReady <-chan struct
 			args = append(args[:i], args[i+1:]...)
 		}
 	}
-
 	after(etcdReady, func() error {
+		etcdNameFile := filepath.Join(s.DataDir, "server", "db", "etcd", "name")
```

Comment:
Maybe make this a local func since we do the same thing with the same magic list of strings so many places.